Test failures with pypy3.10 7.3.13

When running the test suite using PyPy3.10, I'm getting two test failures:

```pytb
$ pypy3 -m pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.10.13[pypy-7.3.13-final], pytest-7.4.3, pluggy-1.3.0
rootdir: /tmp/overrides
plugins: anyio-4.0.0, xprocess-0.23.0, asyncio-0.21.1, xdist-3.4.0
asyncio: mode=strict
collected 67 items                                                                                                                    

tests/test_decorator_params.py .                                                                                                [  1%]
tests/test_enforce__py38.py ...........................                                                                         [ 41%]
tests/test_final.py ........                                                                                                    [ 53%]
tests/test_issubtype_with_typeddict__py38.py .                                                                                  [ 55%]
tests/test_named_and_positional__py38.py ...........                                                                            [ 71%]
tests/test_new_union__py3_10.py ..                                                                                              [ 74%]
tests/test_overrides.py ...FF......                                                                                             [ 91%]
tests/test_signatures.py ...                                                                                                    [ 95%]
tests/test_special_cases.py ...                                                                                                 [100%]

============================================================== FAILURES ===============================================================
___________________________________ OverridesTests.test_overrides_builtin_method_correct_signature ____________________________________

self = <test_overrides.OverridesTests testMethod=test_overrides_builtin_method_correct_signature>

    def test_overrides_builtin_method_correct_signature(self):
>       class SubclassOfInt(int):

tests/test_overrides.py:156: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_overrides.py:158: in SubclassOfInt
    def bit_length(self):
overrides/overrides.py:143: in override
    return _overrides(method, check_signature, check_at_runtime)
overrides/overrides.py:170: in _overrides
    _validate_method(method, super_class, check_signature)
overrides/overrides.py:189: in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
overrides/signature.py:99: in ensure_signature_is_compatible
    same_main_module = _is_same_module(sub_callable, super_callable)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

callable1 = <function OverridesTests.test_overrides_builtin_method_correct_signature.<locals>.SubclassOfInt.bit_length at 0x00007fc1b9dffb00>
callable2 = <function int.bit_length at 0x000055c6c541e180>

    def _is_same_module(callable1: _WrappedMethod, callable2: _WrappedMethod2) -> bool:
        mod1 = callable1.__module__.split(".")[0]
        try:
            mod2 = callable2.__module__
        except AttributeError:
            return False
>       mod2 = mod2.split(".")[0]
E       AttributeError: 'NoneType' object has no attribute 'split'

overrides/signature.py:61: AttributeError
__________________________________ OverridesTests.test_overrides_builtin_method_incorrect_signature ___________________________________

self = <test_overrides.OverridesTests testMethod=test_overrides_builtin_method_incorrect_signature>

    def test_overrides_builtin_method_incorrect_signature(self):
        if sys.version_info >= (3, 7):
            expected_error = self.assertRaises(TypeError)
        else:
            # In 3.6 inspecting signature's isn't possible for builtin
            # methods so this this passes the signature check.
            expected_error = no_error()
    
        with expected_error:
    
>           class SubclassOfInt(int):

tests/test_overrides.py:174: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_overrides.py:176: in SubclassOfInt
    def bit_length(self, _):
overrides/overrides.py:143: in override
    return _overrides(method, check_signature, check_at_runtime)
overrides/overrides.py:170: in _overrides
    _validate_method(method, super_class, check_signature)
overrides/overrides.py:189: in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
overrides/signature.py:99: in ensure_signature_is_compatible
    same_main_module = _is_same_module(sub_callable, super_callable)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def _is_same_module(callable1: _WrappedMethod, callable2: _WrappedMethod2) -> bool:
        mod1 = callable1.__module__.split(".")[0]
        try:
            mod2 = callable2.__module__
        except AttributeError:
            return False
>       mod2 = mod2.split(".")[0]
E       AttributeError: 'NoneType' object has no attribute 'split'

overrides/signature.py:61: AttributeError
======================================================= short test summary info =======================================================
FAILED tests/test_overrides.py::OverridesTests::test_overrides_builtin_method_correct_signature - AttributeError: 'NoneType' object has no attribute 'split'
FAILED tests/test_overrides.py::OverridesTests::test_overrides_builtin_method_incorrect_signature - AttributeError: 'NoneType' object has no attribute 'split'
==================================================== 2 failed, 65 passed in 0.65s =====================================================
```

CC @mattip

right, `overrides` would produce a similar exception on CPython if you make a slightly weird method where you `del` the `__module__` attribute, like this:

```python
class ASuper:
    def some_method(self):
        return 1
    del some_method.__module__
class ASub(ASuper):
    @override
    def some_method(self):
        return 2
```

yields:

```
tests/test_overrides.py:97: in <module>
    class ASub(ASuper):
tests/test_overrides.py:98: in ASub
    @override
overrides/overrides.py:143: in override
    return _overrides(method, check_signature, check_at_runtime)
overrides/overrides.py:170: in _overrides
    _validate_method(method, super_class, check_signature)
overrides/overrides.py:189: in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
overrides/signature.py:99: in ensure_signature_is_compatible
    same_main_module = _is_same_module(sub_callable, super_callable)
overrides/signature.py:61: in _is_same_module
    mod2 = mod2.split(".")[0]
E   AttributeError: 'NoneType' object has no attribute 'split'
```

--- a/overrides/signature.py
+++ b/overrides/signature.py
@@ -54,9 +54,10 @@ def _get_type_hints(callable) -> Optional[Dict]:
 
 def _is_same_module(callable1: _WrappedMethod, callable2: _WrappedMethod2) -> bool:
     mod1 = callable1.__module__.split(".")[0]
-    try:
-        mod2 = callable2.__module__
-    except AttributeError:
+    # "__module__" attribute may be missing in CPython or it can be None
+    # in PyPy: https://github.com/mkorpela/overrides/issues/118
+    mod2 = getattr(callable2, "__module__", None)
+    if mod2 is None:
         return False
     mod2 = mod2.split(".")[0]
     return mod1 == mod2

